lib/spoom/context/sorbet.rb

```diff
@@ -69,7 +69,24 @@ def srb_files(with_config: nil, include_rbis: true)
         allowed_extensions = Spoom::Sorbet::Config::DEFAULT_ALLOWED_EXTENSIONS if allowed_extensions.empty?
         allowed_extensions -= [".rbi"] unless include_rbis
 
-        excluded_patterns = config.ignore.map { |string| File.join("**", string, "**") }
+        excluded_patterns = config.ignore.map do |string|
+          # We need to simulate the behavior of Sorbet's `--ignore` flag.
+          #
+          # From Sorbet docs on `--ignore`:
+          # > Ignores input files that contain the given string in their paths (relative to the input path passed to
+          # > Sorbet). Strings beginning with / match against the prefix of these relative paths; others are substring
+          # > matchs. Matches must be against whole folder and file names, so `foo` matches `/foo/bar.rb` and
+          # > `/bar/foo/baz.rb` but not `/foo.rb` or `/foo2/bar.rb`.
+          string = if string.start_with?("/")
+            # Strings beginning with / match against the prefix of these relative paths
+            File.join(absolute_path, string)
+          else
+            # Others are substring matchs
+            File.join("**", string)
```

Comment:
I wonder if we should add the `absolute_path` here as well. Otherwise, could we not risk matching `string` against a part of the absolute path too? Like, if the absolute path of our project is `/foo/bar/baz` and the ignore was `bar`, `**/bar/**` would match all the files, right?

Reply:
I added it but I'm started to wonder if the `FileCollector` should work with relative paths instead.

At the moment we're making all the paths absolutes, then we visit with `FileCollector`, then we strip the absolute path root from the results. This back and forth seems unnecessary.

Let's see. If the tests pass I can merge this then revisit later.

Reply:
Agreed, doing relative matches would definitely be nicer.